diff --git a/Box2D/CMakeLists.txt b/Box2D/CMakeLists.txt
index a8ad550..17b3069 100644
--- a/Box2D/CMakeLists.txt
+++ b/Box2D/CMakeLists.txt
@@ -26,7 +26,9 @@ if(BOX2D_BUILD_EXAMPLES)
 
   # Testbed and dependencies.
   find_package(OpenGL REQUIRED)
+if(NOT HAIKU) # Haiku has glut, so no need for freeglut
   add_subdirectory(freeglut)
   add_subdirectory(glui)
+endif(NOT HAIKU)
   add_subdirectory(Testbed)
 endif(BOX2D_BUILD_EXAMPLES)
diff --git a/Box2D/Testbed/CMakeLists.txt b/Box2D/Testbed/CMakeLists.txt
index d1be5c0..ad716e4 100644
--- a/Box2D/Testbed/CMakeLists.txt
+++ b/Box2D/Testbed/CMakeLists.txt
@@ -82,11 +82,19 @@ add_executable(Testbed
 	${Testbed_Tests_SRCS}
 )
 
-target_link_libraries (
-	Testbed
-	Box2D
-	freeglut_static
-	glui
-	${ADDITIONAL_LIBRARIES}
-	${OPENGL_LIBRARIES}
-) 
+if(HAIKU) # Haiku has glut, so no need for freeglut
+	target_link_libraries (
+		Testbed
+		Box2D
+		glui
+		${OPENGL_LIBRARIES}
+)
+else(HAIKU)
+	target_link_libraries (
+		Testbed
+		Box2D
+		freeglut_static
+		glui
+		${OPENGL_LIBRARIES}
+)
+endif(HAIKU)
diff --git a/Box2D/Testbed/Framework/Test.cpp b/Box2D/Testbed/Framework/Test.cpp
index 9e6a935..f43ee39 100644
--- a/Box2D/Testbed/Framework/Test.cpp
+++ b/Box2D/Testbed/Framework/Test.cpp
@@ -19,3 +19,6 @@
 #include "Test.h"
 #include <stdio.h>
+#ifdef __HAIKU__
+#include <GL/glut.h>
+#endif
 
diff --git a/Box2D/Testbed/Tests/TestEntries.cpp b/Box2D/Testbed/Tests/TestEntries.cpp
index e12c102..1ca1db1 100644
--- a/Box2D/Testbed/Tests/TestEntries.cpp
+++ b/Box2D/Testbed/Tests/TestEntries.cpp
@@ -18,4 +18,7 @@
 
 #include "../Framework/Test.h"
+#ifdef __HAIKU__
+#include <GL/glut.h>
+#endif
 
 #include "AddPair.h"
